packages/query-core/src/queryObserver.ts

```diff
@@ -683,6 +687,7 @@ function shouldLoadOnMount(
 ): boolean {
   return (
     options.enabled !== false &&
+    options.queryFn !== skipToken &&
```

Comment:
Alternate solution is to assign options.queryFn !== skipToken to options.enabled.
Whenever we initiate or update the observer we would 
```
options.enabled = options.queryFn !== skipToken
```

Downside of this is that we will mutate the options.enabled and if someone is checking it (nobody should do it) it might give him a value that he didn't assign. 
So it's potentially not backwards compatible

Reply:
I think we should centralize the check in the `defaultQueryOptions` function. There, we already set options depending on other options. It would be fine to set `enabled: false` if the `queryFn` is a `skipToken`

Reply:
Done!